lib/socket.js

```diff
@@ -62,20 +61,64 @@ class Transport extends EventEmitter {
   //
   _onSocketData(chunk) {
     const packets = [];
-    this._buffer += chunk;
+    let parsedLength;
+    let newPartLength;
+
+    if (this._buffer) {
+      newPartLength = chunk.indexOf(0);
+      if (newPartLength === -1) {
+        this._buffer = Buffer.concat(
```

Comment:
Oh yeah, we doesn't concat with null, we just assign it to _buffer.

If we assume that 
> There literally can only be two half-messages :)

Then yes, there will be the same amount of copying.
If we don't need to handle the case where there are more that 2 half-messages then my suggestion will only make the code more complex, so we should not implement it in this case.